### PR TITLE
blockdev_inc_backup_merge_bitmaps_with_diff_granularity: Add bitmaps merge test

### PR DESCRIPTION
**Pull request by zhencliu**
Merge two bitmaps with different granularities

Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1909076

### CODE REVIEW

**Comment by zhencliu** [comment]
local fs:
 (1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_merge_bitmaps_with_diff_granularity: PASS (84.83 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-21T02.20-c6739e6/results.html
JOB TIME   : 87.75 s

iscsi direct + virtio-blk:
(1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_merge_bitmaps_with_diff_granularity: PASS (83.61 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-21T03.17-c226c3b/results.html
JOB TIME   : 86.51 s

nbd:
 (1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_merge_bitmaps_with_diff_granularity: PASS (94.36 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-21T03.58-e7a8639/results.html
JOB TIME   : 97.10 s

ceph:
(1/1) type_specific.io-github-autotest-qemu.blockdev_inc_backup_merge_bitmaps_with_diff_granularity: PASS (215.08 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-12-21T04.01-d313e42/results.html
JOB TIME   : 217.79 s

**Comment by zhencliu** [comment]
hi @YongxueHong and @ngu-niny , can you please help to review this patch? Thanks much.

File: qemu/tests/cfg/blockdev_inc_backup_merge_bitmaps_with_diff_granularity.cfg [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    image_backup_chain_data1 = base
+    remove_image_data1 = yes
+    force_create_image_data1 = yes
+    storage_pools = default
+    storage_pool = default
+    storage_type_default = directory
+    full_backup_options = '{"sync": "full"}'
+    bitmap_merge_list = 'bitmap0 bitmap1'
+    bitmap_merge_target = 'bitmap_tmp'
+    granularity_list = 512 1K 2K 4K 8K 16K 32K 64K 128K 256K 512K 1M 2M 4M 8M 16M 32M 64M 128M 256M 512M 1G 2G
```
Comment (YongxueHong) [inline qemu/tests/cfg/blockdev_inc_backup_merge_bitmaps_with_diff_granularity.cfg:25]:
```suggestion
    granularity_list = 512 1K 2K 4K 8K 16K 32K 64K 128K 256K 512K 1M 
    granularity_list += 2M 4M 8M 16M 32M 64M 128M 256M 512M 1G 2G
```
Maybe a bit long.

**Review by YongxueHong — COMMENTED** [review]
LGTM.

**Comment by ngu-niny** [comment]
Test PASS:

10:02:31 INFO | PASS 1-Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_merge_bitmaps_with_diff_granularity

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**